src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/indexed_fields_table/indexed_fields_table.tsx

```diff
@@ -18,45 +18,53 @@
  */
 
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import { createSelector } from 'reselect';
-
+import { IndexPatternField, IIndexPattern } from '../../../../../../../../../plugins/data/public';
 import { Table } from './components/table';
 import { getFieldFormat } from './lib';
+import { IndexedFieldItem } from './types';
 
-export class IndexedFieldsTable extends Component {
-  static propTypes = {
-    fields: PropTypes.array.isRequired,
-    indexPattern: PropTypes.object.isRequired,
-    fieldFilter: PropTypes.string,
-    indexedFieldTypeFilter: PropTypes.string,
-    helpers: PropTypes.shape({
-      redirectToRoute: PropTypes.func.isRequired,
-      getFieldInfo: PropTypes.func,
-    }),
-    fieldWildcardMatcher: PropTypes.func.isRequired,
+interface IndexedFieldsTableProps {
+  fields: IndexPatternField[];
+  indexPattern: IIndexPattern;
+  fieldFilter: string;
+  indexedFieldTypeFilter: string;
```

Comment:
Hm... I see it wasn't a required property before. Please check it again

Reply:
Agree. Marked these properties as not required.